types/node/index.d.ts

```diff
@@ -4431,7 +4535,7 @@ declare module "fs" {
      * If `mode` is a string, it is parsed as an octal integer.
      * If `flag` is not supplied, the default of `'w'` is used.
      */
-    export function writeFileSync(path: PathLike | number, data: any, options?: { encoding?: string | null; mode?: number | string; flag?: string; } | string | null): void;
+    export function writeFileSync(path: PathLike | number, data: BinaryData | string | any, options?: WriteFileOptions): void;
```

Comment:
I don't think the datatype of `data` should be changed here. As `any` is allowed adding others doesn't change anything and doc provides the needed info already.